MiniApp/Classes/core/Storage/MiniAppStatus.swift

```diff
@@ -105,11 +104,11 @@ class MiniAppStatus {
         return storedPermissionsList ?? [:]
     }
 
-    /// Method to delete the existing custom permissions from the Keychain. If a mini app is downloaded for the first time, it is safe to remove the existing custom permission in the keychain.
-    /// - Parameter appId: Mini app ID
-    func validateCustomPermissions(appId: String) {
-        if getMiniAppInfo(appId: appId) == nil {
-            self.miniAppKeyStore.removeKey(for: appId)
+    /// Method to delete the existing custom permissions from the Keychain. Keystore should keep Custom Permissions for only downloaded mini-apps
```

Comment:
Just noting here: We'll also need to change our implementation of how custom permissions are stored. Instead of storing deny/grant for all custom permissions, we should store only the ones in the manifest. i.e. the permissions listed in the settings screen should only list the permissions which are in the Mini App's manifest.

But probably that needs to be in a separate PR when you add caching for the manifest.

Reply:
Yeah, I thought the same. We need to store only the permissions which MiniApp entitled for. Will work on it in the new ticket